src/common/atomic_queue.cpp

```diff
@@ -51,6 +51,8 @@ pthread_cond_t empty[MAX_ATOMIC_THREADS] = {PTHREAD_COND_INITIALIZER};
 pthread_mutex_t pushQmutex[MAX_ATOMIC_THREADS] = {PTHREAD_MUTEX_INITIALIZER};
 pthread_t atid[MAX_ATOMIC_THREADS];
 void *atomicRegionIdRoot;
+std::map<fi_addr_t, fi_addr_t> fiAddrMap;
+pthread_rwlock_t fiAddrLock;
```

Comment:
Initialization of this lock is missing. Initialize this lock at the time of process threads creation.